src/app.md

```diff
@@ -725,6 +726,47 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given an array of unparsed version `&str`s, iterate and filter against a semver `VersionReq` and find the latest `Version`.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::Version;
+use semver::VersionReq;
+use std::str::FromStr;
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::SemVerError);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let mut ver = Version::from_str("0.0.0").unwrap();
+    let vset = ["0.9.0", "1.0.0", "1.0.1"];
+    let r = VersionReq::parse("<= 1.0.0").unwrap();
+
+    for v in vset.iter() {
```

Comment:
Let's try to make this `for` loop more Rust idiomatic by using iterators. For example:
1. `filter_map` to convert valid `vset` string versions to `semver::Version`
2. `filter` versions which does not match your `VersionReq`
3. and use `max` to get the latest version from the range